tests/src/test_cases/test_gridnav.c

```diff
@@ -0,0 +1,113 @@
+#if LV_BUILD_TEST
+#include "../lvgl.h"
+
+#include "unity/unity.h"
+#include "lv_test_indev.h"
+
+static lv_obj_t * g_screen;
+static lv_group_t * g_group;
+static struct {
+    bool press_happened;
+    uint32_t key;
+    lv_obj_t * obj;
+} g_key_data;
+
+void setUp(void)
+{
+    g_screen = lv_screen_active();
+    g_group = lv_group_create();
+    g_key_data.press_happened = false;
+}
+
+void tearDown(void)
+{
+    lv_obj_clean(g_screen);
+    lv_group_delete(g_group); /* also removes all indevs set to the group */
+}
+
+static void key_event_cb(lv_event_t * e)
+{
+    TEST_ASSERT_FALSE(g_key_data.press_happened);
+    g_key_data.press_happened = true;
+    g_key_data.key = *(uint32_t *)lv_event_get_param(e);
+    g_key_data.obj = lv_event_get_target_obj(e);
+}
+
+static void gridnav_one_axis_move_only(uint32_t key_grid_axis_next,
+                                       uint32_t key_grid_axis_prev,
+                                       uint32_t key_obj_axis_next,
+                                       uint32_t key_obj_axis_prev,
+                                       lv_gridnav_ctrl_t gridnav_ctrl,
+                                       lv_flex_flow_t flex_flow)
+{
+    lv_indev_set_group(lv_test_keypad_indev, g_group);
+
+    lv_obj_t * cont = lv_obj_create(g_screen);
+    lv_obj_set_flex_flow(cont, flex_flow);
+    lv_obj_set_flex_align(cont, LV_FLEX_ALIGN_CENTER, LV_FLEX_ALIGN_CENTER, LV_FLEX_ALIGN_CENTER);
+    lv_obj_set_size(cont, lv_pct(100), lv_pct(100));
+    lv_obj_center(cont);
+    lv_gridnav_add(cont, gridnav_ctrl);
+    lv_group_add_obj(g_group, cont);
+    lv_obj_t * objs[3];
+    for(uint32_t i = 0; i < 3; i++) {
+        lv_obj_t * obj = lv_roller_create(cont);
```

Comment:
This is the new test. I could not get the normal `lv_obj` to work instead of a roller. I don't understand what it is about the roller that makes focus and navigate in a gridnav but not a normal `lv_obj`. I looked at the default flags, classes, constructors, etc.

Reply:
Solved it. The obj needed a child in order to be focusable by gridnav.